From 7b1b709abdb995ee5a6f3c56000c01a7eec96fff Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Wed, 13 May 2020 11:00:20 -0400
Subject: [PATCH 1/1] Automatically roll deployments

Automatically roll deployments when config is changed.
Otherwise, the new config is pushed but the behavior
is not. We cannot do the exact thing upstream did
because of an armada bug. We are doing what monitor
does when it runs into the same issue.
Reference upstream commit:
https://github.com/helm/charts/commit/af19146e72a1eae813e7bc6ce21b0aea9cea4341

Signed-off-by: Jerry Sun <jerry.sun@windriver.com>
---
 stable/dex/templates/deployment.yaml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/stable/dex/templates/deployment.yaml b/stable/dex/templates/deployment.yaml
index 5e714e9..1df25b4 100644
--- a/stable/dex/templates/deployment.yaml
+++ b/stable/dex/templates/deployment.yaml
@@ -31,6 +31,8 @@ spec:
       labels:
         app: {{ template "dex.name" . }}
         release: "{{ .Release.Name }}"
+      annotations:
+        configchecksum: {{ toYaml .Values.config | sha256sum | trunc 63 }}
     spec:
       serviceAccountName: {{ template "dex.serviceAccountName" . }}
       nodeSelector:
-- 
2.7.4
